### PR TITLE
fix: trigger on any python source

### PR DESCRIPTION
**Pull request by yuejiaointel**
## Description

This PR changes doc ci behavior to trigger when there are changes to python source files

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

**Comment by yuejiaointel** [comment]
Now doc.yml should be triggered by any change in py files. Added a test in t_sne.py to demonstrate the idea and will remove before merge

**Review by david-cortes-intel — APPROVED** [review]
LGTM, but please remember to remove the test print before merging.

Although it could also be skipped for changes in .py files in tests, by checking only for changes under `sklearnex/` and `daal4py/`.

**Comment by yuejiaointel** [comment]
> LGTM, but please remember to remove the test print before merging.
> 
> Although it could also be skipped for changes in .py files in tests, by checking only for changes under `sklearnex/` and `daal4py/`.

Fixed, now only trigger by python files under these 2 dir

**Review by david-cortes-intel — APPROVED** [review]
